### PR TITLE
#393 Owner permissions for employee resource

### PR DESCRIPTION
**Pull request by vryazanov**
#393 
* owner permissions for employee resource + tests

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#495](https://github.com/timelesslounge/timelessis/issues/495) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#495](https://github.com/timelesslounge/timelessis/issues/495) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/f649a0f3-c52c-438f-b82a-b712adc8549e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by vryazanov** [comment]
@emilianodellacasa my PR assigned to me. I think I can find and leave 3 comments for myself :D
so should I? :dancer:

**Comment by emilianodellacasa** [comment]
@vryazanov that is weird! Open an issue on zerocracy for this

**Comment by emilianodellacasa** [comment]
@0crat assign @marceloamadeu

**Comment by 0crat** [comment]
@0crat[/z](https://www.0crat.com/u/0crat) I see that you unassigned this issue; the order is still assigned to @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
> @0crat assign @marceloamadeu  ([here](https://github.com/timelesslounge/timelessis/issues/495#issuecomment-470680486))

@emilianodellacasa This pull request [#495](https://github.com/timelesslounge/timelessis/issues/495) is assigned to @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/08a426de-0ec2-487a-a34a-1051e43e7dca); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

File: tests/integration/it_owner_privileges_test.py [hunk 1]
```diff
@@ -146,132 +144,32 @@ def test_can_not_manage_locations_from_different_company(clean_app, db_session):
...
+def test_can_manage_employees_from_same_company(method, clean_app):
+    me = factories.EmployeeFactory(company=factories.CompanyFactory())
+    colleague = factories.EmployeeFactory(company=me.company)
+    flask.g.user = me
+
+    assert has_privilege(
+        method=method, resource="employee", employee_id=colleague.id)
+
+
+@pytest.mark.parametrize('method', (
```
Comment (marceloamadeu) [inline tests/integration/it_owner_privileges_test.py]: @vryazanov Also use double quotes.

File: tests/integration/it_owner_privileges_test.py [hunk 2]
```diff
@@ -146,132 +144,32 @@ def test_can_not_manage_locations_from_different_company(clean_app, db_session):
...
+    Method.READ,
+    Method.CREATE,
+    Method.UPDATE,
+    Method.DELETE,
+))
+def test_can_manage_employees_from_same_company(method, clean_app):
+    me = factories.EmployeeFactory(company=factories.CompanyFactory())
+    colleague = factories.EmployeeFactory(company=me.company)
+    flask.g.user = me
+
```
Comment (marceloamadeu) [inline tests/integration/it_owner_privileges_test.py]: @vryazanov This blank line is not required. Can delete?

**Comment by marceloamadeu** [comment]
@vryazanov ping

**Comment by vryazanov** [comment]
@paulodamaso fixed

**Comment by emilianodellacasa** [comment]
@vryazanov you meant @marceloamadeu :-)

**Comment by vryazanov** [comment]
> @vryazanov you meant @marceloamadeu :-)

ooops :)
@marceloamadeu fixed, pls have a look )

**Comment by marceloamadeu** [comment]
@vryazanov Thanks

**Comment by marceloamadeu** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 41 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#495](https://github.com/timelesslounge/timelessis/issues/495) is not in the agenda of @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu), can't inspect

**Comment by 0crat** [comment]
The job [#495](https://github.com/timelesslounge/timelessis/issues/495) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@marceloamadeu According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only cosmetic issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@marceloamadeu ping

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @marceloamadeu[/z](https://www.0crat.com/u/marceloamadeu)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**